.evergreen/install_c_driver.sh

```diff
@@ -1,127 +1,133 @@
-#!/bin/bash
-# Usage examples:
-# MONGOC_VERSION=1.22.1 MONGOCRYPT_VERSION=1.5.2 ./install.sh
-# PREFIX=/tmp/installdir MONGOC_VERSION=1.22.1 MONGOCRYPT_VERSION=1.5.2 ./install.sh
+#!/usr/bin/env bash
 
 set -o errexit
 set -o pipefail
 
-print_usage() {
-    echo "usage: MONGOC_VERSION=<version> MONGOCRYPT_VERSION=<version> ./install.sh"
-}
+declare -r mongoc_version="${mongoc_version:-"${mongoc_version_default:?"missing mongoc version"}"}"
 
-if [[ -z $MONGOC_VERSION ]]; then print_usage; exit 2; fi
-if [[ -z $MONGOCRYPT_VERSION ]]; then print_usage; exit 2; fi
+# Usage:
+#   to_windows_path "./some/unix/style/path"
+#   to_windows_path "/some/unix/style/path"
+to_windows_path() {
+  cygpath -aw "${1:?"to_windows_path requires a path to convert"}"
+}
 
-VERSION=$MONGOC_VERSION
-PREFIX=${PREFIX:-$(pwd)"/../mongoc/"}
-OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+declare mongoc_dir
+mongoc_dir="$(pwd)/mongoc"
 
-if [ "$BSON_EXTRA_ALIGNMENT" = "1" ]; then
-    ENABLE_EXTRA_ALIGNMENT="ON"
+# "i" for "(platform-)independent".
+declare mongoc_idir mongoc_install_idir
+if [[ "${OSTYPE:?}" == "cygwin" ]]; then
+  # CMake requires Windows paths for configuration variables on Windows.
+  mongoc_idir="$(to_windows_path "${mongoc_dir}")"
+  mongoc_install_idir="$(to_windows_path "${mongoc_dir}")"
 else
-    ENABLE_EXTRA_ALIGNMENT="OFF"
+  mongoc_idir="${mongoc_dir}"
+  mongoc_install_idir="${mongoc_dir}"
 fi
-CMAKE_ARGS="
-  -DENABLE_AUTOMATIC_INIT_AND_CLEANUP=OFF -DENABLE_SHM_COUNTERS=OFF
-  -DENABLE_TESTS=OFF -DENABLE_EXAMPLES=OFF -DENABLE_STATIC=ON
-  -DENABLE_EXTRA_ALIGNMENT=${ENABLE_EXTRA_ALIGNMENT} -DCMAKE_MACOSX_RPATH=ON
-  -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_PREFIX_PATH=$PREFIX"
+: "${mongoc_idir:?}"
+: "${mongoc_install_idir}"
```

Comment:
Assuming the intent is to error if this is unset.
```suggestion
: "${mongoc_install_idir:?}"
```